lib/bank_account.ex

```diff
@@ -8,31 +10,78 @@ defmodule BankAccount do
   """
   @opaque account :: pid
 
+  @impl true
```

Comment:
```suggestion
  @impl GenServer
```
The impl tag can be upgraded to specify the actual behaviour that is being implemented. The benefit is less pronounced in things like GenServer where the functions are known, but when you start getting in to self defined behaviours, or multiple behaviours on the same module, it is a good practice to get in to.